### ISSUE TITLE
Crashes when generating keys

### ISSUE THREAD

**Issue by emimeswift** [christophhagen/CryptoKit25519#2]
Hello, I tried to include CryptoKit25519 in a project that needs to support iOS 11+, but it seems to (randomly) crash whenever I generate keys (at least on iOS 13.4 where I am testing).
Here is the simplest sample project I could think of to reproduce the issue (using SwiftUI to be quicker):

```
import SwiftUI
import CryptoKit25519

struct ContentView: View {
    
    @State var privateKeyString = "n/a"
    @State var publicKeyString = "n/a"
    
    var body: some View {
        VStack {
            Text("Private key: \(privateKeyString)")
            Text("Public key: \(publicKeyString)")
            Button("Generate keys") {
                self.generateKeys()
            }
        }
    }
    
    func generateKeys() {
        let privateKey = try! Curve25519.Signing.PrivateKey()
        let privateData = privateKey.rawRepresentation
        privateKeyString = privateData.base64EncodedString()
        
        let publicKey = privateKey.publicKey
        let publicData = publicKey.rawRepresentation
        publicKeyString = publicData.base64EncodedString()
    }
    
}
```

Sometimes (usually - but not always - on the first try), keys are generated and displayed. But most often (almost always as of the second try) the app crashes. Unfortunately the error displayed is not always the same nor is very descriptive.

- Sometimes I get something like

> Thread 1: EXC_BREAKPOINT (code=1, subcode=0x1e378bfa4)

on the row `return Data(bytes)` in the `rawRepresentation` variable in PrivateKey.swift

- Other times I get:

> Thread 3: EXC_BAD_ACCESS (code=1, address=0xf5d65481e50b506b)

on the queue CA DispatchGroup

- When I am "luckiest" I get an EXC_BREAKPOINT exception on the declaration of the AppDelegate class together with the most descriptive error:
```
CryptoTest(589,0x1050ae040) malloc: Heap corruption detected, free list is damaged at 0x2834a8d40
*** Incorrect guard value: 11272132526359900924
CryptoTest(589,0x1050ae040) malloc: *** set a breakpoint in malloc_error_break to debug
```
which would indicate something like... a memory leak perhaps?

There seem to be no discernible pattern among the various kinds of errors. Also, the situation is the same if I remove the part on the publicKey and just generate the privateKey.

Do you perhaps know what could be going on here? Thanks a lot!

**Comment by christophhagen** [comment]
Thanks for raising this issue! Oddly I haven't experienced this error when using the library myself.

I was able to reproduce the issue with the code you provided. I noticed that the code already crashes when only using:
````swift
let privateKey = try! Curve25519.Signing.PrivateKey()
privateKeyString = "\(privateKey)"
````

There seems to be a memory issue when invoking the `ed25519_create_keypair` function. I'll look into it over the next few days and get back to you.

**Comment by christophhagen** [comment]
I think I found the culprit:
When I internally use the `ed25519_create_keypair` function, it calculates a 32-byte public and 32-byte private key from the 32-byte seed, so all my supplied buffers had 32-byte. However, the function internally writes 64 bytes into the buffer for the private key.

I've updated my buffer size accordingly, and pushed a new release. Hopefully this issue is resolved now.

Damn these C libraries for never providing any documentation :)
